src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -281,16 +278,8 @@ public override async Task ProvideCompletionsAsync(CompletionContext completionC
                 {
                     var syntaxContext = await GetOrCreateContextAsync(document, position, cancellationToken).ConfigureAwait(false);
 
-                    var regularItems = await GetItemsWorkerAsync(completionContext, syntaxContext, document, position, options, preselect: false, telemetryCounter, cancellationToken).ConfigureAwait(false);
+                    var regularItems = await GetItemsWorkerAsync(completionContext, syntaxContext, document, position, options, telemetryCounter, cancellationToken).ConfigureAwait(false);
                     completionContext.AddItems(regularItems);
-
-                    // We may or may not want to provide preselected items based on context. For example, C# avoids provide providing preselected items when
-                    // triggered via text insertion in an argument list.
-                    if (await ShouldProvidePreselectedItemsAsync(completionContext, syntaxContext, document, position, options).ConfigureAwait(false))
-                    {
-                        var preselectedItems = await GetItemsWorkerAsync(completionContext, syntaxContext, document, position, options, preselect: true, telemetryCounter, cancellationToken).ConfigureAwait(false);
-                        completionContext.AddItems(preselectedItems);
-                    }
```

Comment:
this is hte core change in the PR.  Instead of a two-pass approach, we just have GetItemsWorkerAsync return all the data we need.